### ISSUE TITLE
status() returns Err: unable to parse whole input

### ISSUE THREAD

**Issue by timlyo** [danburkert/procinfo-rs#20]
I've run into this error in one of my own projects. Downloading the repo and running `cargo test` results in `test pid::status::tests::test_status` failing with the result:

`thread 'pid::status::tests::test_status' panicked at 'called 'Result::unwrap()' on an `Err` value: Error { repr: Custom(Custom { kind: InvalidInput, error: StringError("unable to parse whole input, remaining: Ok(\"NoNewPrivs:\\t0\\nSeccomp:\\t0\\nCpus_allowed:\\tff\\nCpus_allowed_list:\\t0-7\\nMems_allowed:\\t00000000,00000001\\nMems_allowed_list:\\t0\\nvoluntary_ctxt_switches:\\t32\\nnonvoluntary_ctxt_switches:\\t7\\n\")") }) }', /checkout/src/libcore/result.rs:859`
```
$cargo -V: 
cargo 0.18.0 (fe7b0cdcf 2017-04-24)
$rustc -V
rustc 1.17.0 (56124baa9 2017-04-24)
```

**Comment by timlyo** [comment]
The issue seems to be that my `/proc/*/status` contains a `NoNewPrivs:     0` line. This doesn't seem to be present in the data supplied to `test_parser_status` ad when I add it in the same place I get the same error.

Here's my `cat /proc/self/status` for reference
```
Name:   cat
Umask:  0022
State:  R (running)
Tgid:   24829
Ngid:   0
Pid:    24829
PPid:   16368
TracerPid:      0
Uid:    1000    1000    1000    1000
Gid:    100     100     100     100
FDSize: 64
Groups: 7 10 14 54 90 91 92 95 100 994
NStgid: 24829
NSpid:  24829
NSpgid: 24829
NSsid:  16368
VmPeak:     5980 kB
VmSize:     5980 kB
VmLck:         0 kB
VmPin:         0 kB
VmHWM:       676 kB
VmRSS:       676 kB
RssAnon:              64 kB
RssFile:             612 kB
RssShmem:              0 kB
VmData:      312 kB
VmStk:       136 kB
VmExe:        32 kB
VmLib:      1792 kB
VmPTE:        32 kB
VmPMD:        12 kB
VmSwap:        0 kB
HugetlbPages:          0 kB
Threads:        1
SigQ:   1/31262
SigPnd: 0000000000000000
ShdPnd: 0000000000000000
SigBlk: 0000000000000000
SigIgn: 0000000000000000
SigCgt: 0000000000000000
CapInh: 0000000000000000
CapPrm: 0000000000000000
CapEff: 0000000000000000
CapBnd: 0000003fffffffff
CapAmb: 0000000000000000
NoNewPrivs:     0
Seccomp:        0
Cpus_allowed:   ff
Cpus_allowed_list:      0-7
Mems_allowed:   00000000,00000001
Mems_allowed_list:      0
voluntary_ctxt_switches:        0
nonvoluntary_ctxt_switches:     1
```

**Comment by danburkert** [comment]
Thanks for the report!  This typically happens on new and/or strange kernels.  What's the output of `uname -r`, and what distro are you running?  If you are interested in fixing, you can look at 412a5d6 as an example where a new Status line is added.

**Comment by danburkert** [comment]
Yep, looks like this was introduced in the 4.10 line, I've only tested up to 4.9.  Should be an easy field to add, it appears that it should be a `bool`, since it's just a 0/1 flag. 
 https://www.kernel.org/doc/Documentation/prctl/no_new_privs.txt.

**Comment by timlyo** [comment]
```
$uname-r
4.10.11-1-ARCH
```

I'm running Antergos. I've got some experience with Nom, I'll give it a shot later.

**Comment by danburkert** [comment]
This should be fixed in 0.4.0.

**Comment by dralley** [comment]
@danburkert I've got this same problem on Fedora 30

```
$uname -r
5.1.19-300.fc30.x86_64
```

```
Name:	ibus-dconf
Umask:	0002
State:	S (sleeping)
Tgid:	2178
Ngid:	0
Pid:	2178
PPid:	2174
TracerPid:	0
Uid:	1000	1000	1000	1000
Gid:	1000	1000	1000	1000
FDSize:	64
Groups:	10 18 1000 
NStgid:	2178
NSpid:	2178
NSpgid:	2174
NSsid:	1954
VmPeak:	  443408 kB
VmSize:	  443408 kB
VmLck:	       0 kB
VmPin:	       0 kB
VmHWM:	    7060 kB
VmRSS:	    6028 kB
RssAnon:	     632 kB
RssFile:	    5396 kB
RssShmem:	       0 kB
VmData:	   25568 kB
VmStk:	     132 kB
VmExe:	      16 kB
VmLib:	    5248 kB
VmPTE:	      88 kB
VmSwap:	       0 kB
HugetlbPages:	       0 kB
CoreDumping:	0
THP_enabled:	1
Threads:	4
SigQ:	0/77278
SigPnd:	0000000000000000
ShdPnd:	0000000000000000
SigBlk:	0000000000000000
SigIgn:	0000000001001000
SigCgt:	0000000180000000
CapInh:	0000000000000000
CapPrm:	0000000000000000
CapEff:	0000000000000000
CapBnd:	0000003fffffffff
CapAmb:	0000000000000000
NoNewPrivs:	0
Seccomp:	0
Speculation_Store_Bypass:	thread vulnerable
Cpus_allowed:	f
Cpus_allowed_list:	0-3
Mems_allowed:	00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000000,00000001
Mems_allowed_list:	0
voluntary_ctxt_switches:	17
nonvoluntary_ctxt_switches:	29
```

It looks like perhaps it's the ```Speculation_Store_Bypass:	thread vulnerable``` line
